apps/src/templates/VersionHistory.jsx

```diff
@@ -62,10 +66,31 @@ module.exports = React.createClass({
     this.setState({versions: null});
   },
 
+  onConfirmClearPuzzle: function () {
+    this.setState({confirmingClearPuzzle: true});
+  },
+
+  onCancelClearPuzzle: function () {
+    this.setState({confirmingClearPuzzle: false});
+  },
+
+  onClearPuzzle: function () {
+    this.props.handleClearPuzzle();
+    this.props.handleCloseDialog();
+  },
+
   render: function () {
     var versionList;
+    if (this.state.confirmingClearPuzzle) {
+      versionList = (
+        <div>
+          <p>Are you sure you want to clear all Big Game progress&#63;</p>
+          <button id="confirm-button" style={{float: 'right'}} onClick={this.onClearPuzzle}>Start Over</button>
+          <button id="again-button" onClick={this.onCancelClearPuzzle}>Cancel</button>
+        </div>
+      );
```

Comment:
This isn't really a versionList at this point. Wonder if it makes sense to have separate vars, or a name that better encompasses both usages.